src/main/scala/beam/agentsim/agents/parking/ChoosesParking.scala

```diff
@@ -55,10 +69,89 @@ object ChoosesParking {
         driverId = driver.toString
       )
       eventsManager.processEvent(parkEvent) // nextLeg.endTime -> to fix repeated path traversal
+      restOfTrip.foreach { legs =>
+        if (legs.size >= 2 && legs(0).beamLeg.mode == BeamMode.CAR && legs(1).beamLeg.mode == BeamMode.WALK) {
+          val parkingSkimmerEvent = createParkingSkimmerEvent(tick, tazTreeMap, nextActivity, stall, legs)
+          eventsManager.processEvent(parkingSkimmerEvent)
+          val freightRequestType =
+            nextActivity.flatMap(activity =>
+              Option(activity.getAttributes.getAttribute(FREIGHT_REQUEST_TYPE)).asInstanceOf[Option[FreightRequestType]]
+            )
+          freightRequestType.foreach { requestType =>
+            val freightSkimmerEvent = createFreightSkimmerEvent(tick, trip, parkingSkimmerEvent, requestType)
+            eventsManager.processEvent(freightSkimmerEvent)
+          }
+        }
+      }
     }
     currentBeamVehicle.setReservedParkingStall(None)
   }
 
+  private def createFreightSkimmerEvent(
+    tick: Int,
+    trip: Option[EmbodiedBeamTrip],
+    parkingSkimmerEvent: ParkingSkimmerEvent,
+    requestType: FreightRequestType
+  ) = {
+    val (loading, unloading) = requestType match {
+      case FreightRequestType.Unloading => (0, 1)
+      case FreightRequestType.Loading   => (1, 0)
+    }
+    val costPerMile = trip
+      .map { trip =>
+        val carDistanceInM = trip.beamLegs.collect {
+          case leg if leg.mode == BeamMode.CAR => leg.travelPath.distanceInM
+        }.sum
+        logger.info("Cost: {}, dist: {}", trip.costEstimate, carDistanceInM)
+        trip.costEstimate / carDistanceInM * 1609.0
+      }
+      .getOrElse(Double.NaN)
+    val freightSkimmerEvent = new FreightSkimmerEvent(
+      tick,
+      parkingSkimmerEvent.tazId,
+      loading,
+      unloading,
+      costPerMile,
+      parkingSkimmerEvent.walkAccessDistanceInM,
+      parkingSkimmerEvent.parkingCostPerHour
+    )
+    freightSkimmerEvent
+  }
+
+  private def createParkingSkimmerEvent(
+    tick: Int,
+    tazTreeMap: TAZTreeMap,
+    nextActivity: Option[Activity],
+    stall: ParkingStall,
+    restOfTrips: List[EmbodiedBeamLeg]
+  ): ParkingSkimmerEvent = {
+    val walkLeg = restOfTrips(1)
```

Comment:
There is no guarantee that there is a second item. How about `restOfTrips.lift(1)`, which will return `Option[EmbodiedLeg]`

Reply:
Or if it is important, then we should be more explicit in the throwing of an exception